packages/shared/lib/src/sdk/mod.rs

```diff
@@ -380,6 +382,27 @@ impl Sdk {
 
         Ok(())
     }
+
+    // Sign arbitrary data with the provided signing key
+    // TODO: Use function from SDK when it is available
+    pub async fn sign_arbitrary(
+        &self,
+        signing_key: String,
+        data: String,
+    ) -> Result<JsValue, JsError> {
+        let hash = Hash::sha256(&data);
+        let secret = ed25519::SecretKey::from_str(&signing_key)?;
+        let signature: ed25519::Signature = ed25519::SigScheme::sign(&secret, &hash);
+
+        let sig_bytes: &[u8] = &signature.0.to_bytes();
+
+        to_js_result((
+            // Hash
+            &hash.to_string().to_lowercase(),
```

Comment:
Just another question, but why does the hash get returned here?

Reply:
This is useful for the backend - the signature is actually for this hash, so the backend can validate that it was both hashed and signed correctly. And Gianmarco requested that it be in hex-lowercase